dhall/src/Dhall/Util.hs

```diff
@@ -81,3 +90,35 @@ insert expression =
 -- | Prefix used for error messages
 _ERROR :: IsString string => string
 _ERROR = "\ESC[1;31mError\ESC[0m"
+
+get :: (String -> Text -> Either ParseError a) -> Bool -> Maybe FilePath -> IO a
+get parser censor maybeFile = do
+    inText <- do
+        case maybeFile of
+            Just file -> Data.Text.IO.readFile file
+            Nothing   -> Data.Text.IO.getContents
+
+    let name =
+            case maybeFile of
+                Just file -> file
+                Nothing   -> "(stdin)"
+
+    let result = parser name inText
+
+    let censoredResult =
+            if censor then first Dhall.Parser.censor result else result
+
+    Dhall.Core.throws censoredResult
+
+-- | Convenient utility for retrieving an expression
+getExpression
+    :: Bool
+    -- ^ Censor the source code?
+    -> Maybe FilePath
```

Comment:
`Nothing` arguments also smell a bit like boolean blindness to me. IMHO the straightforward fix would be to have both `getExpressionFromFile` and `getExpressionFromStdin`, but it's not important to me.